charts/caddy-ingress-controller/templates/deployment.yaml

```diff
@@ -86,8 +86,8 @@ spec:
             initialDelaySeconds: 3
             periodSeconds: 10
             httpGet:
-              port: 9765
-              path: /metrics
+              port: 80
+              path: /healthz
```

Comment:
I don't like the idea of having the health check on the 80 port (which serve all requests). It would be better to add it either on the metric endpoint or on a new http server

Reply:
If you have it on the metrics endpoint, you wouldn't be able to turn off metrics without breaking the health check. So a separate port might be better.

Reply:
IIRC the metrics server is always on but the metrics handler is enabled only when needed. It should be fine to replace the "static_response" handler by an healthz handler

Reply:
https://github.com/caddyserver/ingress/blob/master/internal/caddy/global/metrics.go#L28 This line would currently disable the whole server if metrics are turned off. I think I encountered this when I tried disabling metrics and my readiness checks would keep failing.

Reply:
Let me try to port it to the metrics endpoint and see if disabling the metrics server still works. Will report back

Reply:
Briefly skim the code. It's definitely doable. I will update the PR to accomodate the change.

I have a question though. Would there be possibility where the `ingress_server` server is down, but the `metrics_server` is still active? This will translate to the controller showing as healthy but can't actually route any traffics.

Reply:
@Embraser01 I've moved the `/healthz` to the metrics server.

 And add a test  to validate the change by comparing the generated Caddy JSON config. Currently, it only has one base test case, and I'm not sure how to add more

Reply:
> I have a question though. Would there be possibility where the ingress_server server is down, but the metrics_server is still active? This will translate to the controller showing as healthy but can't actually route any traffics.

I don't think it can happen, at least I think on config reload, Caddy works on a "all or nothing" way where if the `ingress_server` is not yet started, `metrics_server` will not serve either although I'm not sure (@mholt should know more on this)